src/test/java/com/microsoft/jenkins/azuread/integrations/casc/ConfigAsCodeCertificateTest.java

```diff
@@ -0,0 +1,126 @@
+        assertTrue("authenticated can read", azureAdMatrixAuthorizationStrategy.hasExplicitPermission(PermissionEntry.user(TEST_UPN), Jenkins.READ));
+        assertTrue("authenticated can build", azureAdMatrixAuthorizationStrategy.hasExplicitPermission(PermissionEntry.group("authenticated"), Item.BUILD));
+        assertTrue("authenticated can delete jobs", azureAdMatrixAuthorizationStrategy.hasExplicitPermission(PermissionEntry.user(TEST_UPN), Item.DELETE));
+        assertTrue("authenticated can administer", azureAdMatrixAuthorizationStrategy.hasExplicitPermission(PermissionEntry.user(TEST_UPN), Jenkins.ADMINISTER));
+
+        assertEquals("no warnings", 0, l.getMessages().size());
+
+        {
+            Node agent = j.jenkins.getNode("agent");
+            assertThat(agent, is(notNullValue()));
+            assertThat(agent.getDisplayName(), is(equalTo("agent")));
+            AzureAdAuthorizationMatrixNodeProperty nodeProperty =
+                    agent.getNodeProperty(AzureAdAuthorizationMatrixNodeProperty.class);
+            assertThat(nodeProperty, is(notNullValue()));
+            assertThat(nodeProperty.getInheritanceStrategy(), instanceOf(NonInheritingStrategy.class));
+            assertThat(
+                    nodeProperty
+                            .hasExplicitPermission(PermissionEntry.user("Adele Vance (be674052-e519-4231-b5e7-2b390bff6346)"),
+                                    Computer.BUILD),
+                    is(true)
+            );
+        }
+
+        {
+            Folder folder = (Folder) j.jenkins.getItem("generated");
+            assertNotNull(folder);
+            AzureAdAuthorizationMatrixFolderProperty property = folder.getProperties()
+                    .get(AzureAdAuthorizationMatrixFolderProperty.class);
+            assertTrue("folder property inherits", property.getInheritanceStrategy() instanceof NonInheritingStrategy);
+            String groupSid = "Some group (7fe913e8-6c9f-40f8-913e-7178b7768cc5)";
+            assertTrue(property.hasExplicitPermission(PermissionEntry.group(groupSid), Item.BUILD));
+            assertTrue(property.hasExplicitPermission(PermissionEntry.group(groupSid), Item.READ));
+            assertFalse(property.hasExplicitPermission(PermissionEntry.user("anonymous"), Item.READ));
+            assertTrue(property.hasExplicitPermission(PermissionEntry.group(groupSid), Item.CONFIGURE));
+            assertTrue(property.hasExplicitPermission(PermissionEntry.group(groupSid), Item.DELETE));
+
+            String userSid = "c411116f-cfa6-472c-8ccf-d0cb6053c9aa";
+            assertTrue(property.hasExplicitPermission(PermissionEntry.user(userSid), Item.BUILD));
+        }
+    }
+
+    @Test
+    public void export_configuration() throws Exception {
+        ConfiguratorRegistry registry = ConfiguratorRegistry.get();
+        ConfigurationContext context = new ConfigurationContext(registry);
+        CNode yourAttribute = getJenkinsRoot(context).get("authorizationStrategy");
```

Comment:
this doesn't seem right, its just a complete duplicate of the secret test, for it to assert something of value it should be getting the `securityRealm` and then checking a different export file

Reply:
I did some more improvements, please review